### PR TITLE
NPratt: add internal shared api key env var + api base urls

### PR DESCRIPTION
**Pull request by npratt-plextrac**
### [CTEAM-56](https://plextrac.atlassian.net/browse/CTEAM-56)

**Related PRs:**
- backend: https://github.com/PlexTrac/product-core-backend/pull/6335
- product-deploy-manifests: https://github.com/PlexTrac/product-deploy-manifests/pull/258
- infra-saltstack: https://github.com/PlexTrac/infra-saltstack/pull/553

**Related Ops ticket:**
- https://github.com/PlexTrac/cloudenablement-docs/issues/480

[CTEAM-56]: https://plextrac.atlassian.net/browse/CTEAM-56?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/_podman.sh [hunk 1]
```diff
@@ -178,6 +178,7 @@ function plextrac_start_podman() {
     serviceNames=("plextracdb" "postgres" "redis" "ckeditor-backend" "plextracapi" "notification-engine" "notification-sender" "contextual-scoring-service" "migrations" "plextracnginx")
   fi
   serviceValues[notification-env_vars]="-e API_INTEGRATION_AUTH_CONFIG_NOTIFICATION_SERVICE=${API_INTEGRATION_AUTH_CONFIG_NOTIFICATION_SERVICE:?err}"
+  serviceValues[notification-env_vars]="-e INTERNAL_API_KEY_SHARED=${INTERNAL_API_KEY_SHARED:?err}"
```
Comment (npratt-plextrac) [inline src/_podman.sh:181]: @runia1 does this change make sense to you?

File: static/docker-compose.yml [hunk 2]
```diff
@@ -5,41 +5,40 @@ services:
       update_config:
         parallelism: 1
     depends_on:
-    - plextracdb
-    - redis
-    - postgres
+      - plextracdb
```
Comment (npratt-plextrac) [inline static/docker-compose.yml:8]: looks like my format on save put some work in on this file.. I can revert the formatting changes if this is a concern

File: static/docker-compose.yml [hunk 3]
```diff
@@ -203,15 +204,17 @@ services:
       REDIS_PASSWORD: "${REDIS_PASSWORD:?err}"
       serviceConfig: ${serviceConfig:-}
       API_INTEGRATION_AUTH_CONFIG_NOTIFICATION_SERVICE: "${API_INTEGRATION_AUTH_CONFIG_NOTIFICATION_SERVICE:?err}"
+      INTERNAL_API_KEY_SHARED: "${INTERNAL_API_KEY_SHARED:?err}"
+      CORE_API_BASE_URL: "${CORE_API_BASE_URL:?err}"
```
Comment (runia1) [inline static/docker-compose.yml:208]: non-whitespace

File: static/docker-compose.yml [hunk 4]
```diff
@@ -5,41 +5,40 @@ services:
...
+      - plextracdb
+      - redis
+      - postgres
     environment:
       STARTUP_MODE: API_ONLY
       CLOUD_STORAGE_ENDPOINT: ${CLOUD_STORAGE_ENDPOINT:-minio}
       CLOUD_STORAGE_SSL: ${CLOUD_STORAGE_SSL:-false}
     env_file:
-    - .env
+      - .env
```
Comment (runia1) [inline static/docker-compose.yml:16]: the core-backend will get the new env vars via this.  Not ideal, but it is what it is.

**Review by runia1 — COMMENTED** [review]
I think this mostly makes sense to me.  But the event-orchestrator is not in `docker-compose.yaml` so we'll have to supply the new env-vars to that service [here](https://github.com/PlexTrac/infra-saltstack/blob/55e873993d43684031ee696ba3512d70be9307c0/pillar/role/prod/plextrac-instances/config.sls#L108) I think.

File: static/docker-compose.yml (hunk 2 above)
Comment (StevenPaugh) [reply static/docker-compose.yml:8]: Can we remove the formatting here? I'm slightly concerned the indent will cause issues with file

File: src/_podman.sh (hunk 1 above)
Comment (StevenPaugh) [reply src/_podman.sh:181]: This should be fine

**Review by StevenPaugh — APPROVED** [review]
:shipit:

**Pull request merged**